mozilla-services/socorro
# fix bug 1346891: betaversionrule

This:

1. creates a webapp API called `VersionString` that returns ... well, ... version strings
2. creates a new `ExpiringCache` that has a ttl in addition to max size because we didn't have one and I wasn't enamored with the ones I saw (though cachetools might be a good choice)
3. rewrites the `BetaVersionRule` to use the new cache and new `VersionString` API

To test:

1. go to `http://localhost:8000/api/` and test the `VersionString` API using product/version/builds
2. set `processor.version_string_api` to `http://webapp:8000/api/VersionString/` in your `my.env` file, then process a recent crash from the beta channel with a build that's in your db and make sure it gets the right version
